autometa/taxonomy/ncbi.py

```diff
@@ -408,23 +423,31 @@ def is_valid_taxid(self, taxid):
             ) > 1:
                 raise ValueError(f"taxid contains invalid character(s)! Given: {taxid}")
             taxid = float(taxid)
-        # a boolean check is needed as something can be a boolean but not necessarily a str, eg. TRUE/FALSE
+        # a boolean check is needed as they will evaluate silently to 0 or 1 when cast as ints. FALSE=0, TRUE=1
         # float(taxid).is_integer() checks if it is something like 12.0 vs. 12.9
         # is_integer only takes float as input else raises error, thus isinstance( ,float) is used before it to make sure a float is being passed
         if isinstance(taxid, bool) or (
             isinstance(taxid, float) and not taxid.is_integer()
         ):
-            raise ValueError(f"taxid must be a positive integer! Given: {taxid}")
+            raise ValueError(f"taxid must be an integer! Given: {taxid}")
         taxid = int(taxid)
         if taxid <= 0:
             raise ValueError(f"Taxid must be a positive integer! Given: {taxid}")
-        if not self.names.get(taxid) and not self.nodes.get(taxid):
-            new_taxid = self.merged.get(taxid, taxid)
-            if not self.names.get(new_taxid) and not self.nodes.get(new_taxid):
-                logger.warning(
-                    f"Taxid : {taxid} not found in either nodes.dmp or names.dmp !"
-                )
-            return int(new_taxid)
+        # Checking databases
+        if taxid not in self.names and taxid not in self.nodes and taxid in self.merged:
+            taxid = self.merged[taxid]
+            if taxid not in self.names and taxid not in self.nodes:
+                err_message = f"NCBI databases out of sync. Merged taxid ({taxid}) not found in nodes.dmp or names.dmp!"
+                logger.warning(err_message)
+                raise DatabaseOutOfSyncError(err_message)
+        elif (
+            taxid not in self.names
+            and taxid not in self.nodes
+            and taxid not in self.merged
+        ):
+            err_message = f"NCBI databases out of sync. {taxid} not in found in any of the NCBI databases - nodes.dmp, names.dmp and merged.dmp."
+            logger.warning(err_message)
+            raise DatabaseOutOfSyncError(err_message)
```

Comment:
You could remove some redundancy here:

```suggestion
        if taxid not in self.names and taxid not in self.nodes:
            if taxid not in self.merged:
                err_message = f"NCBI databases out of sync. {taxid} not in found in any of the NCBI databases - nodes.dmp, names.dmp and merged.dmp."
                logger.warning(err_message)
                raise DatabaseOutOfSyncError(err_message)
            else:
                taxid = self.merged[taxid]
                if taxid not in self.names and taxid not in self.nodes:
                    err_message = f"NCBI databases out of sync. Merged taxid ({taxid}) not found in nodes.dmp or names.dmp!"
                    logger.warning(err_message)
                    raise DatabaseOutOfSyncError(err_message)
```